homeassistant/components/konnected/config_flow.py

```diff
@@ -290,14 +322,13 @@ def __init__(self):
         the connection.
         """
         if user_input is None:
-            # update an existing config entry if host info changes
-            entry = await self.async_set_unique_id(
-                self.data[CONF_ID], raise_on_progress=False
-            )
+            # abort config flow if one is already started
+            entry = await self.async_set_unique_id(self.data[CONF_ID])
             if entry and (
                 entry.data[CONF_HOST] != self.data[CONF_HOST]
                 or entry.data[CONF_PORT] != self.data[CONF_PORT]
             ):
+                # update an existing config entry if host info changes
                 entry_data = copy.deepcopy(entry.data)
                 entry_data.update(self.data)
                 self.hass.config_entries.async_update_entry(entry, data=entry_data)
```

Comment:
We can now pass the updates directly to `self._abort_if_unique_id_configured`.

https://github.com/home-assistant/home-assistant/blob/b2d7bc40dc8c054430c6b8e0111be07060db5878/homeassistant/config_entries.py#L820-L833

Reply:
cool - I'll swap over to this